packages/core/src/internal/base/button.base.ts

```diff
@@ -104,23 +118,27 @@ export class CdsBaseButton extends LitElement {
           type="${ifDefined(this.type)}"
         ></button>`;
 
-        this.showClick();
-        this.appendChild(buttonTemplate.getTemplateElement().content.cloneNode(true) as HTMLElement);
-        const button = this.querySelector('.cds-hidden-button') as HTMLButtonElement;
-        button.dispatchEvent(new MouseEvent('click', { relatedTarget: this, composed: true }));
-        button.remove();
+        const marker = document.createElement('div');
+        this.appendChild(marker);
+        render(buttonTemplate, marker);
```

Comment:
This fixes the issue of lit not rendering the type attr from the initial render.